src/templates/finops-hub/modules/dataFactory.bicep

```diff
@@ -314,6 +320,458 @@ resource trigger_exportContainer 'Microsoft.DataFactory/factories/triggers@2018-
+                  type: 'Expression'
+                }
+              }
+            }
+            {
+              name: 'Create or Update Closed Month Export'
+              type: 'WebActivity'
+              dependsOn: [
+                {
+                  activity: 'Set Closed Month Export Name'
+                  dependencyConditions: [
+                    'Succeeded'
+                  ]
+                }
+              ]
+              policy: {
+                timeout: '0.12:00:00'
+                retry: 0
+                retryIntervalInSeconds: 30
+                secureOutput: false
+                secureInput: false
+              }
+              userProperties: []
+              typeProperties: {
+                url: {
+                  value: 'https://@{variables(\'resourceManagementUri\')}@{item().scope}/providers/Microsoft.CostManagement/exports/@{variables(\'exportName\')}?api-version=2021-10-01'
+                  type: 'Expression'
+                }
+                method: 'PUT'
+                headers: {
+                  authorization: {
+                    value: 'Bearer @{activity(\'Get Token\').output.access_token}'
+                    type: 'Expression'
+                  }
+                }
+                body: {
+                  value: '{\n  "properties": {\n    "schedule": {\n      "status": "Active",\n      "recurrence": "Monthly",\n      "recurrencePeriod": {\n        "from": "@{utcNow()}",\n        "to": "2099-12-31T00:00:00Z"\n      }\n    },\n    "partitionData": "True",\n    "format": "Csv",\n    "deliveryInfo": {\n      "destination": {\n        "resourceId": "@{pipeline().parameters.StorageAccountId}",\n        "container": "msexports",\n        "rootFolderPath": "@{item().scope}"\n      }\n    },\n    "definition": {\n      "type": "amortizedcost",\n      "timeframe": "TheLastBillingMonth",\n      "dataSet": {\n        "granularity": "Daily"\n      }\n    }\n  }\n}'
+                  type: 'Expression'
+                }
+              }
+            }
+            {
+              name: 'Set Closed Month Export Name'
+              type: 'SetVariable'
+              dependsOn: [
+                {
+                  activity: 'Create or Update Open Month Export'
+                  dependencyConditions: [
+                    'Succeeded'
+                  ]
+                }
+              ]
+              userProperties: []
+              typeProperties: {
+                variableName: 'exportName'
+                value: {
+                  value: '@if(greater(length(replace(tolower(concat(pipeline().parameters.StorageAccountName,\'camortized\', last(split(item().scope, \'/\')))), \'-\', \'\')), 64), substring(replace(tolower(concat(pipeline().parameters.StorageAccountName,\'camortized\', last(split(item().scope, \'/\')))), \'-\', \'\'), 0, 63), replace(tolower(concat(pipeline().parameters.StorageAccountName,\'camortized\', last(split(item().scope, \'/\')))), \'-\', \'\'))'
```

Comment:
nit: If we do keep separate exports, "c" isn't very clear what it is. Should we explicitly say "closed"? Again, just a nit.

Reply:
Resolved